flink-kubernetes/src/main/java/org/apache/flink/kubernetes/KubernetesClusterDescriptor.java

```diff
@@ -118,7 +118,7 @@ public String getClusterDescription() {
                                         HighAvailabilityServicesUtils.AddressResolution
                                                 .TRY_ADDRESS_RESOLUTION)));
             } catch (Exception e) {
-                client.handleException(e);
+                LOG.error("An exception occurred while instantiating the RestClusterClient.", e);
```

Comment:
Are you sure that we need these log statements? I think this is only required if the `RuntimeException` is nowhere logged.

Reply:
I thought so, too. But some call paths end up forwarding the `RuntimeException` up to the `main` method without any logging. Some paths swallowed the final exception without any logging (happened during closing/cancellation procedures like [LocalExecutor.closeSession(..)](https://github.com/apache/flink/blob/c6997c97c575d334679915c328792b8a3067cfb5/flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/LocalExecutor.java#L239)).
Hence, I stepped back of this approach and for the safer option of logging it as it was discussed in the [initial PR discussion](https://github.com/apache/flink/pull/11427#discussion_r404221581).

Reply:
Hmm, maybe it was the intention to suppress all exceptions while the `LocalExecutor` is being shut down. Also, shouldn't the main thread log an uncaught exception? And if not, then we should probably add it there because who knows which other components follow the pattern to let exceptions bubble up to a point where they can be treated w/o logging it on the way.

The problem with logging exceptions at several places is that the logs become  harder to understand/read.

Reply:
Ok, thanks for clarification. I updated the code accordingly.